src/resources/views/crud/fields/inc/repeatable_row.blade.php

```diff
@@ -27,6 +27,7 @@
                     if(isset($row[$subfield['name']]) || isset($row[$subfield['name'].'[]'])) {
                         $subfield['value'] = $row[$subfield['name']] ?? $row[$subfield['name'].'[]'];
                     }
+                    $subfield['name'] = $field['name'].'['.$repeatable_row_key.']['.$subfield['name'].']';
```

Comment:
Third question. Can't we check if it's an array here?
```php
if (is_string($subfield['name']) {
    $subfield['name'] = $field['name'].'['.$repeatable_row_key.']['.$subfield['name'].']';
} elseif (is_array($subfield['name'])) {
    // probably a date_range but hey, who knows, might be somthing else too
    foreach ($subfield['name'] as $k => $item) {
        $subfield['name'][$k] = $field['name'].'['.$repeatable_row_key.']['.$item['name'].']';
    }
    // and if (for some reason) we can't do that... we can even do something hacky like
    $subfield['some_flag_for_the_field_to_know_its_repeatable'] = true;
}
```

Reply:
Hey @tabacitu 

1 - Can never be. 
2 - If subfield is another repeatable it will have the repeatable name there. 
3 - Not even considering that for now, after the "first repeatable" it's initialized the "repeatable" field name is changed to accomodate "first repeatable row", so imagine the following scenario: 
```php
$this->crud->addFields([
            [
                'name' => 'extras',
                'type' => 'repeatable',
                'fields' => [
                    [
                        'name' => 'testing',
                        'type' => 'repeatable',
                        'fields' => [
                            [
                                'name' => 'rep sub 1',
                            ]
                        ]
                    ],
                    [
                        'name' => 'testing2'
                    ]
                ]
            ],
```
When `extras` initialize, the `testing` repeatable name would become: `extras[row][testing]` breaking the nested repeatable. 

Like you said, I didn't focused on making repeatable work inside repeatable. **It would probably be doable since we now use array names**, but need a big refactor on how repeatable initializes itself and nested fields.

I've fixed the issue with names in JS (where was the problem for regular/relationship repeatables) and added your suggestion to parse the field names when array.

Reply:
Ok, I understand why it wouldn't work. As I said... I still think it's NOT important to support repeatable inside repeatable.

But now that some of the `relationship` fields use `repeatable`... we have to know. What combinations will NOT work? From what I understand... the new relationships that 4.2 brings will not work inside `repeatable` right? Which begs the question... does it make sense to use _any_ relationship field inside `repeatable`??! Not really... right? You only want to use `relationship` to update the relationship on the Model... when you want a relationship inside the repeatable... you're better off using `select_from_array` or `select_from_ajax`, right?

Reply:
If anything, we should _at least_ add them to the "relationship edge cases" issue - https://github.com/Laravel-Backpack/CRUD/issues/4012

Reply:
@tabacitu exactly. 

I don't think there is any sense in using relationships inside repeatable. 

One thing is using repeatable to edit a relation (like we do), other thing is using repeatable field on your own and add relationship fields there. The `CrudField` will be the repeatable, not any of the repeatable fields. 

The only reason people use relationships inside repeatable is because the setup of the fields is easier since relationships support adding only the field name as a relation and backpack does a great job guessing the stuff needed. 

It will show the "relationship" field, but technically it will never be a relationship, it's just a select. 

Like I told you earlier, it's way easier to support repeatable inside repeatable now. I am sure that if I parsed the names a little bit more I would endup making it work ... Maybe without BC, so we can have a look at it in 4.2.xx !